src/main/ruby/truffleruby/core/string.rb

```diff
@@ -1097,6 +1097,74 @@ def rindex(sub, finish = undefined)
     nil
   end
 
+  def byteindex(str, start = 0)
+    is_regex_pattern = Primitive.is_a?(str, Regexp)
+
+    start = Primitive.rb_to_int(start)
+
+    start += bytesize if start < 0
+    if start < 0 || start > bytesize
+      if is_regex_pattern
+        Primitive.regexp_last_match_set(Primitive.caller_special_variables, nil)
+      end
+
+      return nil
+    end
+
+    if start < bytesize && !Primitive.string_is_character_head?(encoding, self, start)
+      raise IndexError, "offset #{start} does not land on character boundary"
+    end
+
+    if is_regex_pattern
+      Primitive.encoding_ensure_compatible(self, str)
+
+      match = Truffle::RegexpOperations.match_from(str, self, start)
+      Primitive.regexp_last_match_set(Primitive.caller_special_variables, match)
+      return match ? Primitive.match_data_byte_begin(match, 0) : nil
+    end
+
+    str = StringValue(str)
+    return start if str.empty?
+    return nil if start + str.bytesize > bytesize
+
+    enc = Primitive.encoding_ensure_compatible_str(self, str)
+    Primitive.string_byte_index(self, str, enc, start)
+  end
+
+  def byterindex(str, finish = bytesize)
+    finish = Primitive.rb_to_int(finish)
+    finish += bytesize if finish < 0
+    return nil if finish < 0
+
+    finish = bytesize if finish > bytesize
```

Comment:
We have `Primitive.min` which seems a nicer way to do this, without calling bytesize twice.

Reply:
Nice, done